lib/pluginmanager/install_strategy/elastic_xpack.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: utf-8
+# In the context of the plugin manager no dependencies are currently loaded.
+require_relative "../../../logstash-core/lib/logstash/version"
+require "pluginmanager/xpack/remote_installer"
+require "pluginmanager/utils/http_client"
+require "pluginmanager/ui"
+require "net/http"
+require "uri"
+
+module LogStash module PluginManager module InstallStrategy
+  class ElasticXpack
```

Comment:
looking at this functionality, I think we could remove this notion of xpack and elasticxpack. all this is is another strategy that takes a name and tries to download it from a "pack repository".

So `InstallStrategy` is essentially a `PackFetchStrategy`:
* `RepositoryStrategy` look for the pack in ELASTIC_XPACK_BASE_URI to fetch the pack
* `UriStrategy` directly use the URI to fetch the pack

And these can yield either a `LocalInstaller` or `RemoteInstaller` strategy. Even here we could remove the notion of XPack and just called it Pack. thoughts?